components/sync15/ios/RustSyncTelemetryPing.swift

```diff
@@ -0,0 +1,622 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+public class RustSyncTelemetryPing {
```

Comment:
Swift might have nice support for deserializable types, what do you think about using something like [`Codable`](https://developer.apple.com/documentation/swift/codable) and [`JSONDecoder`](https://developer.apple.com/documentation/foundation/jsondecoder)?

Reply:
The more I look at it the more I see custom logic that `Codable` won't give us by default... We can implement the `init(from decoder: Decoder)` but I'm not sure that's worth it..